### PR TITLE
Feature/boas 1133 republish message

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

* refactor(api/applications): improve code and structure in exam timetable
* feat(api/applications): record createdAt for exam timetable item
* feat(web/applications): show republish message if items after publish
* fix(web/applications): default to empty array for timetable items

Also refactored some relevant code in the API as I initially intended to put the logic there, but found that half of this work had already been done in the `web` part of the app, so I just made an update there.

Tested locally.

## Issue ticket number and link

[BOAS-1133](https://pins-ds.atlassian.net/browse/BOAS-1133)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

[BOAS-1133]: https://pins-ds.atlassian.net/browse/BOAS-1133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js
Comment (chrisdalykinandcarta) [inline apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:82]: Do you know how we're currently supporting new timetable items being deleted?

File: apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js
Comment (chrisdalykinandcarta) [inline apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js:135]: I thought we were creating individual entries for each line item on creation (rather than just creating the folders), this is what the PINS Data Model expects anyway (it expects a eventLineItems array with unique IDs, and those IDs would be how the front office communicates submissions to us). I guess with this there'd be no concept of line items outside of strings in the description.

Does this mean that for submissions we're looking up the folder by name?

Comment (HadrianHughesKC) [reply apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js:135]: I'm not sure how it works currently, I'd have to dig into the code a bit more. This isn't actually new code, I've just moved it into service from controller. If you look in `examination-timetable-items.controller.js` on `main` the same code is there.

Comment (chrisdalykinandcarta) [reply apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js:135]: Ah apologies I didn't see the delete - I'll move this discussion to the teams chat

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js
Comment (HadrianHughesKC) [reply apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:82]: Do you mean if you delete a timetable item after the timetable is published?

File: apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js
Comment (HadrianHughesKC) [reply apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js:135]: The first commit is all just restructuring. Everything feature-relevant is in the following commits :)

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js
Comment (chrisdalykinandcarta) [reply apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:82]: Yeah I was wondering if we could take the same route instead of recording it on a separate record, but I'll resolve this convo because I don't want to hold up the PR - LGTM!

**Review by francescobrenci-AR — APPROVED** [review]

**Pull request merged**
